pkg/kubectl/cmd/expose.go

```diff
@@ -29,19 +29,23 @@ import (
 const (
 	expose_long = `Take a replicated application and expose it as Kubernetes Service.
 
-Looks up a ReplicationController by name, and uses the selector for that replication controller
-as the selector for a new Service on the specified port.`
+Looks up a replication controller by name, and uses the selector for that replication controller
+as the selector for a new Service on the specified port. You may also use an existing service
+selector.`
 
 	expose_example = `// Creates a service for a replicated nginx, which serves on port 80 and connects to the containers on port 8000.
 $ kubectl expose nginx --port=80 --container-port=8000
 
+// Creates a second service based on the above service, exposing the container port 8443 as port 443 with the name "nginx-https"
+$ kubectl expose service nginx --port=443 --container-port=8443 --service-name=nginx-https
+
 // Create a service for a replicated streaming application on port 4100 balancing UDP traffic and named 'video-stream'.
 $ kubectl expose streamer --port=4100 --protocol=udp --service-name=video-stream`
 )
 
 func (f *Factory) NewCmdExposeService(out io.Writer) *cobra.Command {
 	cmd := &cobra.Command{
-		Use:     "expose NAME --port=port [--protocol=TCP|UDP] [--container-port=number-or-name] [--service-name=name] [--public-ip=ip] [--create-external-load-balancer=bool]",
+		Use:     "expose [RESOURCE] NAME --port=port [--protocol=TCP|UDP] [--container-port=number-or-name] [--service-name=name] [--public-ip=ip] [--create-external-load-balancer=bool]",
```

Comment:
I erred on the side of not breaking the input pattern of existing commands.  We probably want to break CLI before 1.0.  If we don't think a default is valuable let's just change it now.

----- Original Message -----
> Isn't it going to be the default option?  I would say that you expose a
> replication controller by default.  You *can* expose other things.  Although
> I guess a pod is also something you would expose (except you can't expose
> just one safely without doing a label query first).  I'm ok with forcing
> someone to always type RC though.
> 
> ----- Original Message -----
> > >  // Create a service for a replicated streaming application on port 4100
> > >  balancing UDP traffic and named 'video-stream'.
> > >  $ kubectl expose streamer --port=4100 --protocol=udp
> > >  --service-name=video-stream`
> > >  )
> > >  
> > >  func (f *Factory) NewCmdExposeService(out io.Writer) *cobra.Command {
> > >  	cmd := &cobra.Command{
> > > -		Use:     "expose NAME --port=port [--protocol=TCP|UDP]
> > > [--container-port=number-or-name] [--service-name=name] [--public-ip=ip]
> > > [--create-external-load-balancer=bool]",
> > > +		Use:     "expose [RESOURCE] NAME --port=port [--protocol=TCP|UDP]
> > > [--container-port=number-or-name] [--service-name=name] [--public-ip=ip]
> > > [--create-external-load-balancer=bool]",
> > 
> > My take is that if the command is going to be polymorphic to either rc or
> > service, it should require explicitly providing the resource type.
> > Remembering that the resource defaults to rc if omitted is an unnecessary
> > burden for users.
> > 
> > ---
> > Reply to this email directly or view it on GitHub:
> > https://github.com/GoogleCloudPlatform/kubernetes/pull/5977/files#r27234846